standard/statements.md

```diff
@@ -1096,6 +1096,34 @@ is then expanded to:
 
 The variable `e` is not visible to or accessible to the expression `x` or the embedded statement or any other source code of the program. The variable `v` is read-only in the embedded statement. If there is not an explicit conversion ([§10.3](conversions.md#103-explicit-conversions)) from `T` (the iteration type) to `V` (the *local_variable_type* in the `foreach` statement), an error is produced and no further steps are taken.
 
+When the iteration variable is a reference variable (§9.7), a `foreach` statement of the form
+
+```csharp
+foreach (ref V v in x) «embedded_statement»
+```
+
+is then expanded to:
+
+```csharp
+{
+    E e = ((C)(x)).GetEnumerator();
+    try
+    {
+        while (e.MoveNext())
+        {
+            ref V v = ref (V)(T)e.Current;
+            «embedded_statement»
+        }
+    }
+    finally
+    {
+        ... // Dispose e
+    }
+}
+```
+
+The variable `e` is not visible to or accessible to the expression `x` or the embedded statement or any other source code of the program. The reference variable `v` is read-write in the embedded statement, but `v` shall not be ref-reassigned (§12.21.3). If there is not an identity conversion (§10.2.2) from `T` (the iteration type) to `V` (the *local_variable_type* in the `foreach` statement), an error is produced and no further steps are taken.
```

Comment:
Also I will say the first sentence seems a bit over laboured to me. It is of course a copy of line 1097, I just feel that a talented wordsmith could come up with something more concise for both, which escapes me at the moment.